airframe-sql/src/test/scala/wvlet/airframe/sql/analyzer/TypeResolverTest.scala

```diff
@@ -1070,4 +1070,19 @@ class TypeResolverTest extends AirSpec with ResolverTestHelper {
     analyze("select \"prénom\" from (select name as \"prénom\" from A)")
     // No error
   }
+
+  test("resolve CTE in AliasedRelation") {
+    val p1 = analyze("with t1 as (select id from A) select id from (select id from t1) t2")
+    p1.outputAttributes shouldMatch { case List(col: ResolvedAttribute) =>
+      col.fullName shouldBe "id"
+      col.sourceColumn.head.fullName shouldBe "A.id"
+    }
+
+    val p2 = analyze("with t1 as (select id from A) select count(id) from (select id from t1) t2")
+    p2.outputAttributes shouldMatch {
+      case List(SingleColumn(FunctionCall("count", Seq(col: ResolvedAttribute), _, _, _, _), _, _)) =>
+        col.fullName shouldBe "t2.id"
```

Comment:
Without the fix, this was `t1.id` so it couldn't be resolved.

Reply:
I believe the right expectation here is 'id' can be resolved without any qualifier, but the source column is A.id. I guess outputAttributes or some TypeResolver needs to be fixed.

Reply:
I think the correct expectation is: 
```scala
col.fullName shouldBe "id" // `t1.id` is wrongly resolved here
```

Reply:
Checking the resolved plan with trace-level log:
```
wvlet.airframe.sql.analyzer.TypeResolverTest=trace
```

CTERelationRef already adds t1.id as a qualifier, so I think this part is wrong: 
```
[original plan]
[Query]: (id:?) => (?:long)
  [WithQuery]:  => (id:?)
    - ResolvedIdentifier(Id(t1))
    [Project]:  => (id:?)
      - id:? := Id(id)
      [TableRef]
        - A
  [Project]: (id:?) => (?:long)
    - ?:long := FunctionCall(count, Id(id), distinct:false, window:None)
    [AliasedRelation]:  => (id:?)
      - ResolvedIdentifier(Id(t2))
      [Project]:  => (id:?)
        - id:? := Id(id)
        [TableRef]
          - t1

[resolved plan]
[Query]: (id:long) => (?:long)
  [WithQuery]: (id:long, name:string) => (id:long)
    - ResolvedIdentifier(Id(t1))
    [Project]: (id:long, name:string) => (id:long)
      - *id:long <- A.id
      [TableScan] default.A:  => (id:long, name:string)
  [Project]: (id:long) => (?:long)
    - ?:long := FunctionCall(count, *t1.id:long <- A.id, distinct:false, window:None)
    [AliasedRelation]: (id:long) => (id:long)
      - ResolvedIdentifier(Id(t2))
      [Project]: (id:long) => (id:long)
        - *id:long <- A.id
        [CTERelationRef]:  => (id:long)
          - *t1.id:long <- A.id
```

Reply:
After fixing not to add qualifier to cte.outputAttributes, it seems resolveRegularRelation adds unnecessary qualifier:


```
2023-07-11 14:17:32.139-0700 trace [RewriteRule]
transformed with resolveRegularRelation:
[before]
[Project]: (id:?) => (?:long)
  - ?:long := FunctionCall(count, Id(id), distinct:false, window:None)
  [AliasedRelation]: (id:long) => (id:?)
    - ResolvedIdentifier(Id(t2))
    [Project]: (id:long) => (id:?)
      - id:? := Id(id)
      [CTERelationRef]:  => (id:long)
        - *id:long <- A.id

[after]
[Project]: (id:long) => (?:long)
  - ?:long := FunctionCall(count, *t2.id:long <- A.id, distinct:false, window:None)
  [AliasedRelation]: (id:long) => (id:long)
    - ResolvedIdentifier(Id(t2))
    [Project]: (id:long) => (id:long)
      - id:long := *id:long <- A.id
      [CTERelationRef]:  => (id:long)
        - *id:long <- A.id
  - (RewriteRule.scala:32)
  ```

Reply:
Most likely, the current implementation of AliasedRelation.outputAttribuets is wrong, because it adds qualifiers without taking care of the context  https://github.com/wvlet/airframe/blob/dcb0c47375df3e9a50c26d8678361094bfd32092/airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala#L518